### PR TITLE
Fixed cmake linking issues

### PR DESCRIPTION
**Pull request by Umesh-k26**

### CODE REVIEW

**Review by pranav-159 — APPROVED** [review]

File: llvm/lib/Support/CMakeLists.txt [hunk 1]
```diff
@@ -198,4 +198,4 @@ if(LLVM_WITH_Z3)
     )
 endif()
 
-target_link_libraries(LLVMSupport PRIVATE LLVMMLBridge)
```
Comment (svkeerthy) [inline llvm/lib/Support/CMakeLists.txt:201]: Is this change intentional? I think PUBLIC linking is not required in any of the places. Can you please confirm?

Comment (Umesh-k26) [reply llvm/lib/Support/CMakeLists.txt:201]: In this case, it wouldn't be a problem changing it to _PRIVATE_, but it would be, if big changes are there in _LLVMSupport_ which depend on _MLBridge_(since _include_directories_ won't be propagated). 

Also, for dynamic passes, it can linked as  _PRIVATE_. I changed it to _PUBLIC_ to maintain consistency.
